examples/kprobe/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/cilium/ebpf"
+
+	"golang.org/x/sys/unix"
+)
+
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -cflags "$BPF_CFLAGS" -cc clang-10 KProbeExample ./bpf/kprobe_example.c
+
+const Key uint32 = 0
+
+// This program demonstrates how to attach an eBPF program to a kprobe.
+// The program will be attached to the __x64_sys_execve syscall and print out
+// the number of times it has been called every second.
+func main() {
+	stopper := make(chan os.Signal, 1)
+	signal.Notify(stopper, os.Interrupt, syscall.SIGTERM)
+
+	var originalLimit unix.Rlimit
+	// Get the original rlimit values so we can restore them laster.
+	if err := unix.Getrlimit(unix.RLIMIT_MEMLOCK, &originalLimit); err != nil {
```

Comment:
The changed rlimit only applies to the current process (and it's descendants), so there is no need to restore it later.

Reply:
I will change the other example too